manualtesting/manual_testing.go

```diff
@@ -138,18 +138,16 @@ func manualTest(c *api.Context, w http.ResponseWriter, r *http.Request) {
 
 	// Grab the test ID and pick the test
 	testname, ok := params["test"]
-	var err2 *model.AppError
+	if !ok {
+		c.Err = model.NewAppError("/manual", "manaultesting.manual_test.parse.app_error", nil, "", http.StatusBadRequest)
+		return
+	}
+
 	switch testname[0] {
 	case "autolink":
-		err2 = testAutoLink(env)
+		c.Err = testAutoLink(env)
 		// ADD YOUR NEW TEST HERE!
 	case "general":
-		err2 = nil
-	}
-
-	if err != nil {
```

Comment:
Wrong error variable. Should be checking `err2`.

But this error handling is way more complicated than it has to be anyways and we can just delete it all.